### ISSUE TITLE
Investigate a precompiled CSS-in-JS framework vs a static styling framework

### ISSUE THREAD

**Issue by alanbsmith** [Workday/canvas-kit#1854]
## 🔍 Spike /  Exploration

We've talked a lot about moving past Emotion, but we haven't had much direction. It would be helpful to limit the scope of our options and state the specific criteria we're considering.

## Prior Art

- [Breaking up with CSS-in-JS](https://dev.to/srmagura/why-were-breaking-up-wiht-css-in-js-4g9b)
- [Alan's response](https://gist.github.com/alanbsmith/e5b9fad745161e2b61d72dbe0d0cb79c)

## What questions would be answered once this spike was completed?

There are four main topics we need to consider:

- Theme Support
  - We need a theme system we can use robustly throughout Canvas Kit
  - How do we handle theming across multiple CK versions?
- Style Overrides
  - Teams override our styles all the time
  - It is impossible for us to offer CSS variables for everything, so direct overrides of styles is necessary
- Local Scoping
  - Basically naming CSS classnames with hashes to avoid clashing
- Migration Path
  - How do we handle the migration path for teams using Emotion?
  - How can we make it opt-in vs opt-out to reduce friction?

## Acceptance Criteria

- [ ] Explore the options
- [ ] Cover the topics in a GH discussion
- [ ] Share with the team

**Comment by alanbsmith** [comment]
Closing. Nicholas completed this with #2273
